src/gathernodeinputhandler.cpp

```diff
@@ -44,14 +44,6 @@ Status GatherNodeInputHandler::setInput(const std::string& inputName, InferenceE
             return StatusCode::INTERNAL_ERROR;
         }
     } else {
-        auto firstShardTensor = inputsShardsIt->second.begin()->second;
-        if (firstShardTensor->getTensorDesc() != ptr->getTensorDesc()) {
-            SPDLOG_LOGGER_ERROR(dag_executor_logger, "Shard: {} tensor description differ. First shard desc: {}, current shard desc: {}",
```

Comment:
This check needs to be done in the gathering stage -> check if each shard has the same desc.

Reply:
My mistake it is already checked.